### PR TITLE
chore: new contributing path and content for winglang docs 

### PR DESCRIPTION
**Pull request by ainvoner**
separate contributors related docs from users docs in our docs site.
adding a new `contributing` path and moving some content from the docs section to the new path.
to support our docs site automation, also updated the docs directory structure:
`docs/docs` - contain all users related documentation and will be served under winglang.io/docs
`docs/contributing` - contain all contributors related documentation and will be served under winglang.io/contributing  

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: docs/03-language-guide/01-basic-syntax.md
Comment (ekeren) [inline docs/03-language-guide/01-basic-syntax.md:1]: @ainvoner why is this file removed?

Comment (ainvoner) [reply docs/03-language-guide/01-basic-syntax.md:1]: I don't know why it looks like it... but the file exists in it's correct location under docs/docs/03-language-guide/01-...

https://github.com/winglang/wing/blob/contributing-docs/docs/docs/03-language-guide/01-basic-syntax.md

**Review by ekeren — APPROVED** [review]
